### DISCUSSION TITLE
Ctx argument for classNames - undocumented?

### DISCUSSION THREAD

**Discussion by 7iomka** [mantinedev/mantine#4929] [Q&A] [answered]
I rewrite my codebase to mantine v7

Now classNames prop can be function or object.
For cases when you create your own wrapper around mantine component, and to keep both formats, alongside with custom classNames, you need something like

```tsx
    const resolvedPassedClassNames = classNames
      ? typeof classNames === 'function'
        ? classNames(theme, props, null)
        : classNames
      : {};

    const resolvedClassNames = customVariantStyles?.classNames
      ? getResolvedClassNames(customVariantStyles.classNames, resolvedPassedClassNames)
      : classNames;

```
Question is: what is the third parameter for function version of classNames? 
 ctx: unknown.

<img width="818" alt="image" src="https://github.com/mantinedev/mantine/assets/18473137/349fcd97-c2a6-4c3e-8eea-37600c4aa99d">

I passed the null - it is correct?
Seems like this is not documented or am I missing something?

**Comment by rtivital** [comment, marked as answer]
It is a context parameter used rarely in case component CSS variables cannot be determined based on component props alone. You can view Input component source code for an example. If ctx is unknown then you can safely pass null value there.
